src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -553,6 +554,26 @@ def get_group_to_groupevent(
     return group_to_groupevent
 
 
+def get_dcgs_by_group(
+    groups_to_fire: dict[GroupId, set[DataConditionGroup]],
+    event_data: EventRedisData,
+    dcg_type: DataConditionHandler.Group,
+) -> dict[GroupId, set[DataConditionGroup]]:
+    """
+    Extract DataConditionGroups from groups_to_fire, grouped by group ID.
```

Comment:
might be useful to describe how this works, and what `trigger_group_to_dcg_model` is since it's not defined here.